SimPEG/Utils/matutils.py

```diff
@@ -37,6 +40,9 @@ def mkvc(x, numDims=1):
 
 def sdiag(h):
     """Sparse diagonal matrix"""
+    if isinstance(h, Zero):
+        return h
```

Comment:
right, it purposely doesn't have a shape, but is this potentially confusing to read?